### DISCUSSION TITLE
Function parameters syntax edge case

### DISCUSSION THREAD

**Discussion by atisharma** [hylang/hy#2607] [Q&A] [answered]
Not sure if this is a bug or not, perhaps I am being stupid but from the docs I would expect the following to work:

```hylang
=> (defn f [a #* args * [b None]] )
Traceback (most recent call last):
  File "stdin-c8cbf6a41b297a4cc4cd14f6d7b68de32e687b85", line 1
    (defn f [a #* args * [b None]] )
            ^
hy.errors.HySyntaxError: parse error for pattern macro 'defn': got unexpected end of macro call, expected: end of macro call
```

**Comment by Kodiologist** [comment, marked as answer]
> from the docs I would expect the following to work

Can you be more specific about how the manual led you to believe this? `*` after `*args` isn't allowed in Python, either, and it's redundant.

**Reply by atisharma** [reply]
> perhaps I am being stupid

I think this is probably the case - apologies for imposing brain fog on your time. It is indeed redundant.

Although, there's no specific sentence that explicitly precludes it in the Hy docs.

> If the parameter list contains #* args or (unpack-iterable args), then args is set to a tuple containing all otherwise unmatched positional arguments. 

> If the symbol * is given in place of a parameter, it means that all the following parameters can only be set by name.

**Reply by Kodiologist** [reply]
The relevant bit is
> Each of these special constructs… has the same restrictions as in Python
